modules/jdbc/src/main/java/org/apache/ignite/internal/jdbc/JdbcStatement.java

```diff
@@ -396,7 +398,7 @@ public ResultSet getResultSet() throws SQLException {
 
         JdbcResultSet rs = resSets.get(curRes);
 
-        if (!rs.isQuery()) {
+        if (rs == null || !rs.isQuery()) {
```

Comment:
if null is considering as legit element of `resSets` collection, then we need to mark generic type as `@Nullable`

Reply:
getResultSet() marked as nullable, @Nullable JdbcResultSet rs = too, did i miss smth ?

Reply:
i found that resSets has volatile notation seems our Statement implementation is not a thread safe at all.
apeend @Nullable, fix nullability checks

Reply:
should be fixed